### PR TITLE
Fix Pad and Gather incorrect usage of HasType helpers.

### PR DESCRIPTION
**Pull request by edgchen1**
**Description**
Fix Pad and Gather incorrect usage of HasType helpers.

**Motivation and Context**
Fix issues.

### CODE REVIEW

File: onnxruntime/core/providers/cpu/tensor/pad.cc [hunk 1]
```diff
@@ -244,7 +244,7 @@ static Status PadImpl(OpKernelContext* ctx,
                       const std::vector<int64_t>& slices,
                       const Mode& mode,
                       T value) {
-  if (!utils::HasType<AllEnabledPadTypes, T>()) {
+  if (!utils::HasTypeWithSameSize<AllEnabledPadTypes, T>()) {
     return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "Input data type not supported in this build.");
   }
```
Comment (skottmckay) [inline onnxruntime/core/providers/cpu/tensor/pad.cc:250]: Is it better to do this in Compute prior to all the PadImpl calls?

File: onnxruntime/core/providers/cpu/tensor/pad.cc [hunk 2]
```diff
@@ -517,22 +517,15 @@ Status Pad::Compute(OpKernelContext* ctx) const {
     slices_to_use = &slices_;
   }
 
-  Status pad_status{};
   switch (element_size) {
```
Comment (skottmckay) [inline onnxruntime/core/providers/cpu/tensor/pad.cc:520]: nit: it's easier to set a break point if there's just one return point, but I guess you could set that in PadImpl

**Review by skottmckay — APPROVED** [review]
:shipit:

File: onnxruntime/core/providers/cpu/tensor/pad.cc (hunk 2 above)
Comment (edgchen1) [reply onnxruntime/core/providers/cpu/tensor/pad.cc:520]: ok, i can revert this in a future PR

**Pull request merged**